### PR TITLE
feat: Add text highlighting to note bookmarks

### LINKED ISSUE #1342: Add the ability to highlight text within a note
### Describe the feature you'd like

I would like to highlight specific lines within a note, similar to how we can highlight content in a link or an article.

Is this something that has been considered?

### Describe the benefits this would bring to existing Karakeep users

I think it would be nice because I often conduct research in various places, such as ChatGPT or Claude, and come up with comprehensive results, usually in text form. I have been copying this information into a carekeep. However, since it's not technically an article, I cannot highlight it in the same way.

### Can the goal of this request already be achieved via other means?

I can take the text and place it elsewhere so that it can be ingested by Karakheep. It could also be stored and transformed into an art piece, such as a link or an article that can be highlighted. I could try to do something like that as well.

### Have you searched for an existing open/closed issue?

- [x] I have searched for existing issues and none cover my fundamental request

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by AdrianAcala**
Allow users to highlight text within note bookmarks, extending the existing highlighting functionality to markdown content. Users can now select text in notes and apply colored highlights (yellow, red, green, blue) with the same interface used for link content.

This feature addresses the need to emphasize specific parts of copied text from sources like ChatGPT or Claude, improving the research note-taking experience by making content more navigable and emphasizable.

The implementation reuses the existing highlight database schema and tRPC routes, ensuring consistency with link highlighting while adding comprehensive test coverage for text-specific scenarios including unicode characters, overlapping highlights, and edge cases.

Fixes #1342

### CODE REVIEW

File: apps/web/components/dashboard/preview/HighlightsBox.tsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import React from "react";
```
Comment (xuatz) [inline apps/web/components/dashboard/preview/HighlightsBox.tsx]:
```suggestion
import { Fragment } from "react";
```

File: apps/web/components/dashboard/preview/BookmarkTextHighlighter.tsx [hunk 2]
```diff
@@ -0,0 +1,406 @@
+import React, { useEffect, useRef, useState } from "react";
```
Comment (xuatz) [inline apps/web/components/dashboard/preview/BookmarkTextHighlighter.tsx]:
```suggestion
import { useEffect, useRef, useState, FC, ComponentProps, PointerEvent } from "react";
```

**Comment by AdrianAcala** [comment]
Updated @xuatz ! Good catch! Thanks!

**Comment by erauner12** [comment]
So excited for this, thank you!

**Comment by erauner12** [comment]
Following up on this one @AdrianAcala , need any feedback. I have not attempted development on this project yet but let me know if there is anything I can do!

**Comment by AdrianAcala** [comment]
Oh geez. I just saw this. I've been waiting for a review but I realize I wasn't clear that I fixed it needed a new review. Now there's conflicts I've got to work through with failing tests.

**Comment by AdrianAcala** [comment]
Already implemented into the app. Closing PR.
